mne/io/edf/edf.py

```diff
@@ -90,86 +90,58 @@ def __init__(self, input_fname, montage, eog=None, misc=None,
                            "Both 'annot' and 'annotmap' must be specified."))
 
         # Raw attributes
-        last_samps = [self._edf_info['nsamples'] - 1]
+        last_samps = [edf_info['nsamples'] - 1]
         super(RawEDF, self).__init__(
-            info, preload, last_samps=last_samps, orig_format='int',
+            info, preload, filenames=[input_fname], raw_extras=[edf_info],
+            last_samps=last_samps, orig_format='int',
             verbose=verbose)
 
         logger.info('Ready.')
 
-    def __repr__(self):
-        n_chan = self.info['nchan']
-        data_range = self.last_samp - self.first_samp + 1
-        s = ('%r' % os.path.basename(self.info['filename']),
-             "n_channels x n_times : %s x %s" % (n_chan, data_range))
-        return "<RawEDF  |  %s>" % ', '.join(s)
-
     @verbose
-    def _read_segment(self, start=0, stop=None, sel=None, data_buffer=None,
-                      projector=None, verbose=None):
+    def _read_segment_file(self, data, idx, offset, fi, start, stop,
+                           cals, mult):
         """Read a chunk of raw data"""
         from scipy.interpolate import interp1d
-        if sel is None:
-            sel = np.arange(self.info['nchan'])
-        if projector is not None:
-            raise NotImplementedError('Currently does not handle projections.')
-        if stop is None:
-            stop = self.last_samp + 1
-        elif stop > self.last_samp + 1:
-            stop = self.last_samp + 1
-        sel = np.array(sel)
-
-        #  Initial checks
-        start = int(start)
-        stop = int(stop)
-
-        n_samps = self._edf_info['n_samps']
-        buf_len = self._edf_info['max_samp']
+        if mult is not None:
```

Comment:
my understanding of mult in the info is that it is there but it is never used. this is the reason the data is being stored in V instead of uV. Also there is no corresponding unit for uV listed in the constants.